src/main/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStore.kt

```diff
@@ -166,6 +179,205 @@ class ObservationResultsStore(private val dslContext: DSLContext) {
     return ObservationRollupResultsModel.of(plantingSiteId, plantingZoneResults)
   }
 
+  private val biomassAdditionalSpeciesMultiset =
+      with(OBSERVATION_BIOMASS_ADDITIONAL_SPECIES) {
+        DSL.multiset(
+                DSL.select(
+                        SPECIES_ID,
+                        SPECIES_NAME,
+                        IS_INVASIVE,
+                        IS_THREATENED,
+                    )
+                    .from(this)
+                    .where(OBSERVATION_ID.eq(OBSERVATION_BIOMASS_DETAILS.OBSERVATION_ID))
+                    .and(MONITORING_PLOT_ID.eq(OBSERVATION_BIOMASS_DETAILS.MONITORING_PLOT_ID))
+                    .orderBy(SPECIES_ID, SPECIES_NAME))
+            .convertFrom { result ->
+              result.map {
+                BiomassAdditionalSpeciesModel(
+                    isInvasive = it[IS_INVASIVE]!!,
+                    isThreatened = it[IS_THREATENED]!!,
+                    speciesId = it[SPECIES_ID],
+                    speciesName = it[SPECIES_NAME],
+                )
+              }
+            }
+      }
+
+  private val biomassQuadratDetailsMultiset =
+      with(OBSERVATION_BIOMASS_QUADRAT_DETAILS) {
+        DSL.multiset(
+                DSL.select(
+                        POSITION_ID,
+                        DESCRIPTION,
+                    )
+                    .from(this)
+                    .where(OBSERVATION_ID.eq(OBSERVATION_BIOMASS_DETAILS.OBSERVATION_ID))
+                    .and(MONITORING_PLOT_ID.eq(OBSERVATION_BIOMASS_DETAILS.MONITORING_PLOT_ID))
+                    .orderBy(POSITION_ID))
+            .convertFrom { result -> result.map { it[POSITION_ID]!! to it[DESCRIPTION] }.toMap() }
```

Comment:
Could maybe use `result.associate { it[POSITION_ID]!! to it[DESCRIPTION] }` to avoid creating an intermediate list.